features/poller/monitors/core/src/test/java/org/opennms/netmgt/poller/monitors/FtpMonitorTest.java

```diff
@@ -31,57 +31,76 @@
 import java.io.BufferedReader;
 import java.io.InputStreamReader;
 import java.lang.reflect.UndeclaredThrowableException;
+import java.net.InetAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
 import java.net.UnknownHostException;
 import java.util.HashMap;
 import java.util.Map;
 
 import junit.framework.TestCase;
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Ignore;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
 
 import org.opennms.core.utils.InetAddressUtils;
+import org.opennms.core.test.MockLogAppender;
+import org.opennms.core.test.OpenNMSJUnit4ClassRunner;
 import org.opennms.netmgt.poller.PollStatus;
 import org.opennms.netmgt.poller.mock.MockMonitoredService;
+import org.springframework.test.context.ContextConfiguration;
 
-public class FtpMonitorTest extends TestCase {
+@RunWith(OpenNMSJUnit4ClassRunner.class)
+@ContextConfiguration(locations={
+    "classpath:/META-INF/opennms/emptyContext.xml"
+})
+public class FtpMonitorTest {
     private FtpMonitor m_monitor = new FtpMonitor();
     private ServerSocket m_serverSocket = null;
     private Thread m_serverThread = null;
     private static int TIMEOUT = 2000;
 
-    @Override
-    protected void setUp() throws Exception {
-        super.setUp();
-        
+    @Before
+    public void setUp() throws Exception {
         m_serverSocket = new ServerSocket();
         m_serverSocket.bind(null); // don't care what address, just gimme a port
+        MockLogAppender.setupLogging(true);
     }
 
-    @Override
-    protected void tearDown() throws Exception {
+    @After
+    public void tearDown() throws Exception {
         if (m_serverSocket != null && !m_serverSocket.isClosed()) {
             m_serverSocket.close();
         }
         
         if (m_serverThread != null) {
             m_serverThread.join(1500);
         }
-        
-        super.tearDown();
     }
 
+    @Ignore
```

Comment:
Is there a particular reason, why these tests have been disabled?

Reply:
ah yes, forgot to revert that - that test wasn't working on my build machine so I meant to temporarily disable it.

Reply:
actually, these were already being skipped - I converted this test file to use annotations rather than naming conventions & extending TestCase, but they were being skipped before since they depend on external systems.